README
NAME
    Dist::Zilla::Util::Git::Wrapper - Vivify a Git::Wrapper instance for
    Dist::Zilla

VERSION
    version 0.004002

SYNOPSIS
        use Dist::Zilla::Util::Git::Wrapper;

        my $wrapper =  Dist::Zilla::Util::Git::Wrapper->new( zilla => $self->zilla );

        $wrapper->log(); # etc.

    Yes, this does very little, but simply serves as an abstraction for
    getting a Git::Wrapper object from git, mostly, because everyone
    remembers "$self->zilla", but you have to RTFM and go source diving to
    know about "$self->zilla->root"

    This is a simple straight forward way of doing what you want with the
    least mental overhead.

    Everything else is just proxy methods for "git"

    This is a more I.O.C approach to my swathes of
    "Dist::Zilla::Role::Git::LocalRepository" stuff, which got so complex
    with dependency management my head exploded.

ATTRIBUTES
  "zilla"
    Parameter lazily required, and should be a Dist::Zilla object ( or
    compatible )

  "git"
    A Git::Wrapper instance. Optional, and pointless to specify manually,
    but could be useful for testing where you don't want Dist::Zilla

    Vivified if not specified based on the "zilla" parameter.

AUTHOR
    Kent Fredric <kentnl@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2014 by Kent Fredric
    <kentfredric@gmail.com>.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Release history for Dist-Zilla-Util-Git-Wrapper

0.004003 2017-03-07T17:29:30Z 313243a

0.004002 2017-03-07T17:16:31Z 3b203aa
 - Bugfix: Fix test failures with -Ddefault_inc_excludes_dot

 [Dependencies::Stats]
 - Dependencies changed since 0.004001, see misc/*.deps* for details
 - configure: +1 (suggests: +1)
 - develop: +8 ↑4 -2 (recommends: ↓1, suggests: ↑2)
 - runtime: (recommends: +1)
 - test: +1 ↓1 (recommends: +1 ↓1)

0.004001 2014-10-26T09:36:03Z 9dbf8d8
 [Dependencies::Stats]
 - Dependencies changed since 0.004000, see misc/*.deps* for details
 - configure: (recommends: ↑1)
 - develop: +5 ↑1 -1 (recommends: +1, suggests: ↑2)
 - test: (recommends: ↑2)

 [Metadata]
 - Author = kentnl

 [Tests]
 - Avoid locale issues entirely by avoiding them being in regex.
 - This should close #3 ( Thanks Slaven Rezić )

0.004000 2014-08-27T14:16:46Z
 [00 Major]
 - Changes that may affect hand-operated downstreams
 - Tooling Changes.
 - Misc test and metadata fixes.

 [01 Tookit Change]
 - Toolkit switched from Module::Build to EUMM
 - Toolkit upgrade policy softened to recommendations.
 - https://questhub.io/realm/perl/quest/5388cc12bbd0be5b2500011f

 [Dependencies::Stats]
 - Dependencies changed since 0.003000, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +9 ↑1 -8 (suggests: +2 -1)
 - test: +2 ↓1 (recommends: +3)

 [Metadata]
 - Olivier Mengué is now an x_contributor
 - Migrated to kentnl org

 [Tests]
 - Migrate from ReportVersions::Tiny to Test::ReportPrereqs
 - Fix locale specific test ( closes #1, via pull #2, thanks to Olivier Mengué )

0.003000 2013-12-20T13:14:46Z
 - Add UtilRole for easier making of Plugins with a git property.

 [Dependencies::Stats]
 - Dependencies changed since 0.002000, see misc/*.deps* for details
 - runtime: +2 ↑1
 - test: +2

0.002000 2013-12-14T13:15:29Z
 - Make tests pass on Git 1.4
 - Reimplement some parts in terms of UtilRole::MaybeZilla

 [Depedencies::Noteworthy]
 - Drop MooseX::LazyRequire for UtilRole::MaybeZilla
 - add Sort::Versions for Git 1.4 tests

 [Dependencies::Stats]
 - Dependencies changed since 0.001002, see misc/*.deps* for details
 - runtime: +1 -1
 - test: +1

0.001002 2013-12-09T04:09:39Z
 - Report Git Versions
 - Make basic test more resilient

0.001001 2013-12-06T21:46:53Z
 - Make tests more resilient to env indifference.

0.001000 2013-12-06T18:44:46Z
 - First version.


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
use strict;
use warnings;

use Module::Build 0.28;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4208"
  },
  "configure_requires" => {
    "Module::Build" => "0.4208"
  },
  "dist_abstract" => "Vivify a Git::Wrapper instance for Dist::Zilla",
  "dist_author" => [
    "Kent Fredric <kentfredric\@gmail.com>"
  ],
  "dist_name" => "Dist-Zilla-Util-Git-Wrapper",
  "dist_version" => "0.003001",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Util::Git::Wrapper",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Dist::Zilla::UtilRole::MaybeZilla" => 0,
    "Git::Wrapper" => 0,
    "Moose" => 0,
    "Moose::Role" => 0,
    "perl" => "5.008",
    "strict" => 0,
    "utf8" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Log::Contextual::LogDispatchouli" => 0,
    "Log::Dispatchouli" => 0,
    "Path::Tiny" => 0,
    "Sort::Versions" => 0,
    "Test::Fatal" => 0,
    "Test::More" => "1.001003"
  }
);


my %fallback_build_requires = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Log::Contextual::LogDispatchouli" => 0,
  "Log::Dispatchouli" => 0,
  "Module::Build" => "0.4208",
  "Path::Tiny" => 0,
  "Sort::Versions" => 0,
  "Test::Fatal" => 0,
  "Test::More" => "1.001003"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.009.
use strict;
use warnings;

use 5.008;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Vivify a Git::Wrapper instance for Dist::Zilla",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Util-Git-Wrapper",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.008",
  "NAME" => "Dist::Zilla::Util::Git::Wrapper",
  "PREREQ_PM" => {
    "Dist::Zilla::UtilRole::MaybeZilla" => 0,
    "Git::Wrapper" => 0,
    "Moose" => 0,
    "Moose::Role" => 0,
    "strict" => 0,
    "utf8" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Log::Contextual::LogDispatchouli" => 0,
    "Log::Dispatchouli" => 0,
    "Path::Tiny" => 0,
    "Sort::Versions" => 0,
    "Test::Fatal" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.004003",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla::UtilRole::MaybeZilla" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Git::Wrapper" => 0,
  "Log::Contextual::LogDispatchouli" => 0,
  "Log::Dispatchouli" => 0,
  "Moose" => 0,
  "Moose::Role" => 0,
  "Path::Tiny" => 0,
  "Sort::Versions" => 0,
  "Test::Fatal" => 0,
  "Test::More" => 0,
  "strict" => 0,
  "utf8" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
